src/platform/packages/shared/kbn-ebt-tools/src/performance_metrics/context/use_page_ready.ts

```diff
@@ -8,22 +8,60 @@
  */
 
 import { useEffect, useState } from 'react';
+import usePrevious from 'react-use/lib/usePrevious';
 import type { CustomMetrics, Meta } from './performance_context';
 import { usePerformanceContext } from '../../..';
 
-export const usePageReady = (state: {
+interface UsePageReadyProps {
   customMetrics?: CustomMetrics;
   isReady: boolean;
   meta?: Meta;
-}) => {
-  const { onPageReady } = usePerformanceContext();
+  isRefreshing: boolean;
+  customInitialLoadReported?: { value: boolean; onInitialLoadReported: () => void };
+}
 
-  const [isReported, setIsReported] = useState(false);
+export const usePageReady = ({
+  customInitialLoadReported,
+  isReady,
+  isRefreshing,
+  customMetrics,
+  meta,
+}: UsePageReadyProps) => {
+  const { onPageReady, onPageRefreshStart } = usePerformanceContext();
+  const prevIsRefreshing = usePrevious(isRefreshing);
+  const [isInitialLoadReportedInternal, setIsInitialLoadReportedInternal] = useState(false);
+
+  const isInitialLoadReported = customInitialLoadReported
+    ? customInitialLoadReported.value
+    : isInitialLoadReportedInternal;
 
   useEffect(() => {
-    if (state.isReady && !isReported) {
-      onPageReady({ customMetrics: state.customMetrics, meta: state.meta });
-      setIsReported(true);
+    // Skip until either the page is ready for the first time or a refresh cycle begins
+    if (!isInitialLoadReported && !isReady) return;
+
+    // Initial load flow
+    if (isReady && !isInitialLoadReported) {
```

Comment:
nit: How about using `isInitialLoad` instead of `isInitialLoadReported`? (and set it to true by default)
At first glance, it seems we are checking not to be the initial load (because of `!isInitialLoadReported`) which is different from the comment above.